### PR TITLE
Created question models

### PR DESCRIPTION
**Pull request by GavinBurris42**
…ed the region question as an example.

Issue #, if available:

Description of changes: Created question models to update Simple-EC2 UI with BubbleTea. Updated the region question as an example.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/questionModel/keyValue.go
Comment (snay2) [inline pkg/questionModel/keyValue.go:171]: Is it necessary to initialize the model again? If so, can we extract the value of `HeaderStrings` into a constant to avoid shotgun surgery with line 57 if the header names ever change?

File: pkg/questionModel/multiSelectList.go
Comment (snay2) [inline pkg/questionModel/multiSelectList.go:14]: Is this comment correct and necessary?

File: pkg/questionModel/multiSelectList.go
Comment (snay2) [inline pkg/questionModel/multiSelectList.go:60]: Do `keyValue.go` and `plainText.go` also need this window size handling code?

File: pkg/questionModel/multiSelectList.go
Comment (snay2) [inline pkg/questionModel/multiSelectList.go:65]: Does this error message print to the console? If so, I'd prefer something friendlier or matter-of-fact, or something that describes what will happen next (e.g., "Goodbye!" or "Exiting the questionnaire"). Hitting Ctrl+C is not improper behavior for a user, so I don't want the message to sound accusatory.

File: pkg/questionModel/plainText.go
Comment (snay2) [inline pkg/questionModel/plainText.go:27]: Why 156?

File: pkg/questionModel/plainText.go
Comment (snay2) [inline pkg/questionModel/plainText.go:28]: Where does this width value come from?

File: pkg/questionModel/plainText.go
Comment (snay2) [inline pkg/questionModel/plainText.go:88]: Is this TODO still outstanding or have you addressed it?

File: pkg/questionModel/questionModel.go
Comment (snay2) [inline pkg/questionModel/questionModel.go:26]: Can you add an inline comment to each `lipgloss.Color()` statement with the color name?

File: pkg/questionModel/questionModel.go
Comment (snay2) [inline pkg/questionModel/questionModel.go:86]: This looks like it was copied and pasted from someone else's code, which is fine, but we may need to verify that their code is under an acceptable license for us to re-use it in this project. Do you know the source?

File: pkg/questionModel/questionModel.go
Comment (snay2) [inline pkg/questionModel/questionModel.go:160]: Is this comment correct? It seems semantically different than the code.

File: pkg/questionModel/questionModel.go
Comment (snay2) [inline pkg/questionModel/questionModel.go:177]: It might be a good idea to create a constant for the column separator, since it's used in several places.

**Review by snay2 — CHANGES_REQUESTED** [review]
Left some suggestions. We also need to fix the unit tests before we can merge.

File: pkg/questionModel/questionModel.go
Comment (GavinBurris42) [reply pkg/questionModel/questionModel.go:86]: I found this code on the readme for bubble tea. I can change it to be less blatantly obvious

File: pkg/questionModel/plainText.go
Comment (GavinBurris42) [reply pkg/questionModel/plainText.go:88]: I've addressed it

File: pkg/questionModel/plainText.go
Comment (GavinBurris42) [reply pkg/questionModel/plainText.go:28]: I made it up. It was originally 20, but I made it larger because I put in test values and 20 seemed too short

Comment (GavinBurris42) [reply pkg/questionModel/plainText.go:28]: For exmaple an AMI was longer than 20 characters

Comment (GavinBurris42) [reply pkg/questionModel/plainText.go:28]: Im actually removing it because removing it makes it so theres no limit

File: pkg/questionModel/keyValue.go
Comment (GavinBurris42) [reply pkg/questionModel/keyValue.go:190]: Yes. When the user hits enter after typing something in it will go to the next input, or to the add tag button. Also after using the add tag button the cursor will go back to the Key text box

File: pkg/questionModel/questionModel.go
Comment (snay2) [inline pkg/questionModel/questionModel.go:49]: Thanks for adding the color comments on lines 45 and 46. Can you also add one here?

```suggestion
	errorStyle = lipgloss.NewStyle().Foreground(lipgloss.Color("9")) // Red
```

File: pkg/questionModel/questionModel.go
Comment (snay2) [inline pkg/questionModel/questionModel.go:87]: Typo:

```suggestion
// itemDelegate defines how an item is rendered in a list
```

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:148]: You can remove this now too.

```suggestion
```

File: pkg/questionModel/multiSelectList.go
Comment (snay2) [reply pkg/questionModel/multiSelectList.go:83]: What was your decision on this? You removed it from singleSelect, and keyValue and plainText never had it, but multiSelect still does. Just making sure that was intentional.

File: pkg/questionModel/keyValue.go
Comment (snay2) [inline pkg/questionModel/keyValue.go:211]: This function makes the whole file easier to read. Thanks for adding it!

**Review by snay2 — CHANGES_REQUESTED** [review]
A few minor updates, but on the whole this is looking good.

Also, the unit tests still seem to be failing. Any ideas there?

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:258]: Is this a TODO you can address before we merge?

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:230]: I'd prefer to keep this PR as focused as possible. We have three actual question implementations now, so let's save any more for a separate PR.

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:301]: Can you add handling for non-nil values of `err`, like the other two questions have?

**Review by snay2 — CHANGES_REQUESTED** [review]

**Comment by snay2** [comment]
Discussed offline: In the interest of getting a demo-able implementation, we will accept the unit test failures for now while we finish implementing all the question types. Then after that, we will figure out the issue with the interaction between bubble tea and the test driver and fix the unit tests before merging this into the main repo.

We can merge this PR once the remaining two comments are addressed.

File: pkg/ec2helper/ec2helper.go
Comment (snay2) [inline pkg/ec2helper/ec2helper.go:1522]: Can you also add this code to the open [PR to add spot support](https://github.com/awslabs/aws-simple-ec2-cli/pull/85) on the main repo?

File: pkg/questionModel/multiSelectList.go
Comment (snay2) [inline pkg/questionModel/multiSelectList.go:63]: Prefer to rewrite this using active voice:

```suggestion
	m.errorMsg = "Please choose at least one option"
```

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:348]: The default instance type is not necessarily one that the user saved; I'd prefer to phrase it like this:

```suggestion
		{fmt.Sprintf("Use the default instance type, [%s]", *defaultOption)}}
```

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:397]: Typo: `by` -> `be`. But I'd prefer to rewrite with active voice. Some ideas:

* Enter an instance type
* Which instance type should we use?
* Which instance type do you want?

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:425]: Ideas for rewriting the vCPU and memory questions with active voice:

* How many vCPUs do you want?
* How many vCPUs do you need?

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:674]: The different casing styles in these table headers is odd. Perhaps we should standardize on initial capitals ("Profile Name") and then let the table rendering code decide whether to keep that capitalization or make everything uppercase. That way we can change the style all in one place.

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:694]: Typo:

```suggestion
	question := "After how many minutes should the instance terminate? (0 for no auto-termination)"
```

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:1261]: Remove this block

File: pkg/question/question.go
Comment (snay2) [inline pkg/question/question.go:1369]: This might deserve a standalone file in the `questionModel` folder

**Review by snay2 — CHANGES_REQUESTED** [review]
This is a lot of code! 🤣

If you're planning to update the question prompts in a separate PR, feel free to ignore my comments about those for now.

File: cmd/launch.go
Comment (GavinBurris42) [reply cmd/launch.go:438]: The error check on line 437 is more about not going through with the integer conversion if there is an error with the question itself. Do you think it would be better to have this question, and others like it, have the question convert the string to an int and have it return the integer?

File: cmd/launch.go
Comment (GavinBurris42) [reply cmd/launch.go:473]: Yes and no. I added in defaults for the security groups since the question could use the properly now. I made the ReadSubnetPlaceholder function return a boolean again to make it like the other functions since it had error checking now. 

This wont work if added into prod now because I changed how the ReadSecurityGroups method works

File: cmd/launch.go
Comment (snay2) [reply cmd/launch.go:438]: Right, the error check on 437 is good. I'm referring to the fact that we have a new `err` object on line 438 that we never inspect.

But that's a good idea--if we know that the only valid answer for a particular kind of question is a number, we should return it that way at the lowest level possible. However, I'd say save that for a subsequent PR.

File: cmd/launch.go
Comment (snay2) [reply cmd/launch.go:473]: All right, thanks for the explanation.

File: cmd/launch.go
Comment (GavinBurris42) [reply cmd/launch.go:438]: If there is an error caused by lien 438 then it would be caught in line 440. I just left it there cause it wouldnt hurt, but its not really necessary.

File: cmd/launch.go
Comment (GavinBurris42) [reply cmd/launch.go:615]: I addressed this issue with making it return nil instead if its "None" or "" for BootScript. Ill make it return nil for "" with UserTags too. I misconverted the functionality of that line when adding in errors.

File: pkg/cfn/cfn.go
Comment (GavinBurris42) [reply pkg/cfn/cfn.go:56]: When a New VPC is being created it creates a new Cloud Formation stack. If the stack name is the same then it throws an error. This happens when the answer for the AskVPC question is "Create new VPC with default CIDR and %d subnets".

File: pkg/ec2helper/ec2helper.go
Comment (GavinBurris42) [reply pkg/ec2helper/ec2helper.go:1300]: Yep thank u

File: pkg/question/question.go
Comment (GavinBurris42) [reply pkg/question/question.go:397]: I'll address this in the next PR when I reformat the questions

File: pkg/question/question.go
Comment (GavinBurris42) [reply pkg/question/question.go:425]: I'll address this in the next PR when I reformat the questions

File: pkg/question/question.go
Comment (GavinBurris42) [reply pkg/question/question.go:674]: When the headers are created theyre styled with the Bold styling from Lipgloss. This makes them capitalized and bolded.

File: pkg/question/question.go
Comment (GavinBurris42) [reply pkg/question/question.go:1369]: How so? Its just a function to ask the same question instead of rewriting it in multiple places?

Comment (GavinBurris42) [reply pkg/question/question.go:1369]: Its not creating a new question type

File: pkg/ec2helper/ec2helper.go
Comment (GavinBurris42) [reply pkg/ec2helper/ec2helper.go:1522]: How do I do that?

File: pkg/question/question.go
Comment (snay2) [reply pkg/question/question.go:1360]: It's a judgment call either way--go ahead and leave it here for now.

As you say, under the hood it's not a new question type because it wraps singleSelect. But logically it is a distinct kind of question that is used more than once, so I could make an argument that it deserves to be a first-class question type on its own, even though it's just a special case of singleSelect.

File: pkg/ec2helper/ec2helper.go
Comment (snay2) [reply pkg/ec2helper/ec2helper.go:1507]: I left a "suggestion"-style comment on the other PR that includes it. You should be able to accept that suggestion and then test to make sure it still works (e.g., pass an empty region, since that's the error we were seeing yesterday).

https://github.com/awslabs/aws-simple-ec2-cli/pull/85#discussion_r932828364

**Review by snay2 — APPROVED** [review]
LGTM. The only comments that haven't been addressed are unit tests and question names, which will be in future PR(s).

**Pull request merged**
